src/components/table/columns.ts

```diff
@@ -34,6 +39,32 @@ export class ColumnDefinitionFactory {
     }
 }
 
+/**
+ * Formats the header of the column
+ *
+ * @param {Column} column the configuration for the column
+ *
+ * @returns {string | HTMLElement} custom component that renders a column header
+ */
+export const formatHeader = (column: Column) => (): string | HTMLElement => {
+    const element = document.createElement('div');
+    element.innerText = column.title;
+    if (!column.headerComponent) {
+        return;
+    }
```

Comment:
Do we need this check? :thinking: 
We are already checking if the column has specified a header component above in the `create` function